lib/rubocop/runner.rb

```diff
@@ -72,7 +72,13 @@ def inspect_files(files)
 
       each_inspected_file(files) { |file| inspected_files << file }
     ensure
-      ResultCache.cleanup(@config_store, @options[:debug]) if cached_run?
+      # OPTIMIZE: Calling ResultCache.cleanup takes time, so we avoid it when
+      # the inspecting files size is small.
```

Comment:
This working is a bit better

```suggestion
      # inspecting only one file.
```

Reply:
Thank you!!

Reply:
Squashed.  Co-Authored-By: @connorshea . https://github.com/rubocop-hq/rubocop/pull/7069/commits/46680ddddaf18175b845ca07d87fe38009b8aacf